### ISSUE TITLE
Fix dead code Scala compiler warnings

### ISSUE THREAD

**Issue by jbaiera** [elastic/elasticsearch-hadoop#972]
Subtask of #969 

Dead code should be removed to clear the existing warnings out.

```
/Users/james.baiera/Documents/source/elasticsearch-hadoop/spark/sql-20/src/main/scala/org/elasticsearch/spark/sql/DataFrameValueWriter.scala:112: dead code following this construct
    value match {
          ^
/Users/james.baiera/Documents/source/elasticsearch-hadoop/spark/sql-20/src/main/scala/org/elasticsearch/spark/sql/DataFrameValueWriter.scala:112: dead code following this construct
    value match {
    ^
```

**Comment by takezoe** [comment]
These warnings has been fixed in #940.

**Comment by jbaiera** [comment]
@takezoe Indeed they were. Looks like I was running this on a branch that didn't have #940. Closing this.
